src/main/java/io/aiven/kafka/connect/transforms/ExtractTopicFromSchemaName.java

```diff
@@ -0,0 +1,96 @@
+ */
+
+package io.aiven.kafka.connect.transforms;
+
+import java.util.Map;
+import java.util.Optional;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.apache.kafka.common.config.ConfigDef;
+import org.apache.kafka.connect.connector.ConnectRecord;
+import org.apache.kafka.connect.errors.DataException;
+import org.apache.kafka.connect.transforms.Transformation;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public abstract class ExtractTopicFromSchemaName<R extends ConnectRecord<R>> implements Transformation<R> {
+
+    private static final Logger log = LoggerFactory.getLogger(ExtractTopicFromSchemaName.class);
+
+    private Map<String, String> schemaNameToTopicMap;
+    private Pattern pattern;
+
+    @Override
+    public ConfigDef config() {
+        return ExtractTopicFromSchemaNameConfig.config();
+    }
+
+    @Override
+    public void configure(final Map<String, ?> configs) {
+        final ExtractTopicFromSchemaNameConfig config = new ExtractTopicFromSchemaNameConfig(configs);
+        schemaNameToTopicMap = config.schemaNameToTopicMap();
+        final Optional<String> regex = config.regEx();
+        regex.ifPresent(s -> pattern = Pattern.compile(s));
+    }
+
+    @Override
+    public R apply(final R record) {
+
+        if (null == record.valueSchema() || null == record.valueSchema().name()) {
+            throw new DataException(" value schema name can't be null: " + record);
+        }
+        // First check schema value name -> desired topic name mapping and use that if it is set.
+        if (schemaNameToTopicMap.containsKey(record.valueSchema().name())) {
+            return createConnectRecord(record, schemaNameToTopicMap.get(record.valueSchema().name()));
+        }
+        // Secondly check if regex parsing from schema value name is set and use that.
+        if (pattern != null) {
+            final Matcher matcher = pattern.matcher(record.valueSchema().name());
+            if (matcher.find() && matcher.groupCount() == 1) {
+                return createConnectRecord(record, matcher.group(1));
+            }
+            log.trace("No match with pattern {} from schema name {}", pattern.pattern(), record.valueSchema().name());
+        }
+        // If no other configurations are set use value schema name as new topic name.
+        return createConnectRecord(record, record.valueSchema().name());
+    }
+
+    private R createConnectRecord(final R record, final String newTopicName) {
+        return record.newRecord(
+                newTopicName,
+                record.kafkaPartition(),
+                record.keySchema(),
+                record.key(),
+                record.valueSchema(),
+                record.value(),
+                record.timestamp(),
+                record.headers()
+        );
+    }
+
+    @Override
+    public void close() {
+    }
+
+    public static class Value<R extends ConnectRecord<R>> extends ExtractTopicFromSchemaName<R> {
```

Comment:
Sorry but that's not what I meant, not just renaming. If we create a `Value` class  then the logic related to extracting necessary data from value should go to this class in this casse it can be soemthign like 
```
@Override
protected String getSchemaName(final R record) {
    if (null == record.valueSchema() || null == record.valueSchema().name()) {
        throw new DataException(" value schema name can't be null: " + record);
    }
    return record.valueSchema().name();
}
```
in this class, abstract method in super class and similar implementation for `Key` if I understand the logic of this transformation correctly. In the end it will be possible to use this method in `apply` to extract the name of the schema.